### PR TITLE
Actual Pseudo data test

### PR DESCRIPTION
**Pull request by Emilorzz**

### CODE REVIEW

File: src/shared/assertExtensions.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#include <assert.h>
```
Comment (frederikja163) [inline src/shared/assertExtensions.h:1]: also need header macros for this file

File: src/tests/fileIOSuite.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+#endif
+
+#include <string.h>
+
+#include "CuTest.h"
+#include "fileIO.h"
+
+#define TEST_STRING "Lorem ipsum dolor sit amet, consectetur adipiscing elit.\nMorbi dignissim nunc vel lectus egestas, nec rutrum nibh gravida.\n\t(){}[];:<>\\\"'"
+
+/* When writing to a file and reading the file to a string, it should be the same */
```
Comment (frederikja163) [inline src/tests/fileIOSuite.c:16]: I would either remove these comments, or rewrite them a bit to be clearer and distribute them to the methods testing each file.

File: src/tests/fileIOSuite.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+#endif
+    fclose(filePtr);
+}
+
+void test_read_string_size(CuTest *tc) {
+    FILE *filePtr;
+    char *read_string;
+    size_t fileSize, readSize;
+
+    if ((filePtr = fopen("src/tests/testString3.tmp", "r")) == NULL) {
```
Comment (frederikja163) [inline src/tests/fileIOSuite.c:74]: an assert would be justified here

File: src/tests/fileIOSuite.c [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        fprintf(stderr, "Error : could not open file");
+        exit(EXIT_FAILURE);
+    }
+
+    fseek(filePtr, 0, SEEK_END);
+    fileSize = ftell(filePtr);
+    fseek(filePtr, 0, SEEK_SET);
+
+    if ((read_string = malloc((fileSize + 1) * sizeof(char))) == NULL) {
```
Comment (frederikja163) [inline src/tests/fileIOSuite.c:83]: an assert here would be great too

File: src/tests/fileIOSuite.c [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        fprintf(stderr, "Error : could not allocate memory");
+        exit(EXIT_FAILURE);
+    }
+
+#ifdef _WIN32
+    _setmode(_fileno(filePtr), _O_BINARY);
+#endif
+    if ((readSize = fread(read_string, sizeof(char), fileSize, filePtr)) != fileSize) {
```
Comment (frederikja163) [inline src/tests/fileIOSuite.c:91]: assert here

File: src/tests/fileIOSuite.c [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    CuAssertTrue(tc, fileSize == strlen(string));
+
+#ifdef _WIN32
+    _setmode(_fileno(filePtr), _O_BINARY);
+#endif
+    fclose(filePtr);
+}
+
+void test_read_string_size(CuTest *tc) {
```
Comment (frederikja163) [inline src/tests/fileIOSuite.c:69]: For this method i would write the file as the first thing OR use the same file for all your methods.
It is unclear to me why this is the only test that doesn't write its own file. However you could totally just write one file and test that file throughout all the tests.

However; if you decide to only write one file make sure to write a comment in the get_file_io_suite function explaining that the order is important to ensure a file is written before it is read.

**Review by frederikja163 — CHANGES_REQUESTED** [review]
Looks good, but a few changes are necessary. If you dont understand any of my comments feel free to chat to me on discord or write a comment. Also all of the test methods should be marked as static.

File: src/shared/assertExtensions.h (hunk 1 above)
Comment (Emilorzz) [reply src/shared/assertExtensions.h:1]: header macro added

File: src/tests/fileIOSuite.c (hunk 2 above)
Comment (Emilorzz) [reply src/tests/fileIOSuite.c]: The comments have been rewritten these comments and placed them by their respective functions

File: src/tests/fileIOSuite.c (hunk 4 above)
Comment (Emilorzz) [reply src/tests/fileIOSuite.c]: assert added

File: src/tests/fileIOSuite.c (hunk 5 above)
Comment (Emilorzz) [reply src/tests/fileIOSuite.c]: assert added

File: src/tests/fileIOSuite.c (hunk 6 above)
Comment (Emilorzz) [reply src/tests/fileIOSuite.c:63]: made the function independent by writing a file in the start of the function which is then used to test.

**Review by frederikja163 — APPROVED** [review]
Looks good to me. Nice addition with our first tests to the library

File: src/tests/fileIOSuite.c [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+/* Test to check that a file is created on the specified path */
+void test_write_file_exists(CuTest *tc) {
+    const char *string = TEST_STRING;
+    FILE *filePtr;
+
+    write_file("src/tests/testString2.tmp", string);
+    CuAssertTrue(tc, (filePtr = fopen("src/tests/testString2.tmp", "r")) != NULL);
+
+    fclose(filePtr);
+}
```
Comment (Aavild) [inline src/tests/fileIOSuite.c:39]: What if the file already exists? Then this would not test anything

**Review by Aavild — APPROVED** [review]

File: src/tests/fileIOSuite.c (hunk 7 above)
Comment (Emilorzz) [reply src/tests/fileIOSuite.c:39]: well no, but actually yes

**Pull request merged**